intake_stac/catalog.py

```diff
@@ -347,67 +323,71 @@ def stack_bands(self, bands):
                     f'{band} not found in list of eo:bands in collection.'
                     f'Valid values: {sorted(list(set(valid_band_names)))}'
                 )
-
-            value = assets.get(band)
-            band_type = value.get('type')
-            types.append(band_type)
-
-            href = value.get('href')
-            pattern = href.replace(band, '{band}')
-            if 'path_as_pattern' not in item:
-                item['path_as_pattern'] = pattern
-            elif item['path_as_pattern'] != pattern:
-                raise ValueError(
-                    f'Stacking failed: {href} does not contain '
-                    'band info in a fixed section of the url'
```

Comment:
previous implementation assumes asset URLs are compatible with the path_as_pattern extraction of coordinate values from the URL. I don't think we should require that. The downside is that you end up with repeating coordinate values by default (e.g. band = [1,1,1] since xr.open_rasterio() defaults to opening each file with band = [1])

Reply:
Is there any way to intuit the band numbering or ordering. If so, we can always fill in the coordinate values after the fact.

Reply:
@jhamman this is where I get confused. Ultimately we create an intake entry and then delegate opening to intake-xarray plugin (https://github.com/intake/intake-xarray/blob/master/intake_xarray/raster.py). if  or just returns an xarray dataset with bands = [1,1,1...] etc. Here is what the user interface currently looks like:

```python
list_of_bands = ['B1', 'B2']
stack = item.stack_bands(list_of_bands)
da = stack().to_dask()
da['band'] = list_of_bands 
da
```
question for @martindurant how to specify the coordinates of the returned xarray dataarray? Unless we can override driver methods in this library, if I understand correctly we'd need to add a kwarg here https://github.com/intake/intake-xarray/blob/1883e30a0faab6a07e7e522bf45bc96e80f62f2c/intake_xarray/raster.py#L60

Reply:
Yes, I think that's right.
intake-xarray has emerged as a sort of glue with various options, so it would be inkeeping with current evolution; but we might want to have a chat more generally about what intake-xarray is really trying to achieve. I mean, one might delegate *all* kwargs to xarray, or one might try to build out a set of helper methods and extra logic that only make sense at this slightly higher level. We currently do a bit of both.